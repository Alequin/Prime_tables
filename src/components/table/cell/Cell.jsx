import React from "react"

class Cell extends React.Component{

  constructor(props){
    super(props)
    this.onClick = this.onClick.bind(this)
  }

  onClick(){
    alert(this.props.info)
  }

  render(){
    return (
      <div className="cell-container" onClick={this.onClick}>
        {this.props.value}
      </div>
    )
  }
}

export default Cell
