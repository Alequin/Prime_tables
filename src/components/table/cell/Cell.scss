.cell-container{
  text-align: center;
  width: 50px;
  padding: 5px;
  font-size: 18px;
  border: 1px solid black;
}
