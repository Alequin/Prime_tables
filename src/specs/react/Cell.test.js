import React from 'react';
import Cell from './../../components/table/cell/Cell.jsx';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Cell value={5} info={"I am info"}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
