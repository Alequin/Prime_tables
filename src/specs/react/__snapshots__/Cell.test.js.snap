// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cell-container"
  onClick={[Function]}
>
  5
</div>
`;
